source/Core/Language.php

```diff
@@ -991,37 +991,43 @@ protected function _getLanguageFileData($blAdmin = false, $iLang = 0, $aLangFile
     /**
      * Returns language map array
      *
-     * @param int  $iLang   language index
-     * @param bool $blAdmin admin mode [default NULL]
+     * @param int  $languageIndex language index
+     * @param bool $isAdmin       admin mode [default NULL]
      *
      * @return array
      */
-    protected function _getLanguageMap($iLang, $blAdmin = null)
+    protected function _getLanguageMap($languageIndex, $isAdmin = null)
     {
-        $blAdmin = isset($blAdmin) ? $blAdmin : $this->isAdmin();
-        $sKey = $iLang . ((int) $blAdmin);
-        if (!isset($this->_aLangMap[$sKey])) {
-            $this->_aLangMap[$sKey] = array();
-            $myConfig = $this->getConfig();
-
-            $sMapFile = '';
-            $sParentMapFile = $myConfig->getAppDir() . '/views/' . ($blAdmin ? 'admin' : $myConfig->getConfigParam("sTheme")) . '/' . oxRegistry::getLang()->getLanguageAbbr($iLang) . '/map.php';
-            $sCustomThemeMapFile = $myConfig->getAppDir() . '/views/' . ($blAdmin ? 'admin' : $myConfig->getConfigParam("sCustomTheme")) . '/' . oxRegistry::getLang()->getLanguageAbbr($iLang) . '/map.php';
-
-            if (file_exists($sCustomThemeMapFile) && is_readable($sCustomThemeMapFile)) {
-                $sMapFile = $sCustomThemeMapFile;
-            } elseif (file_exists($sParentMapFile) && is_readable($sParentMapFile)) {
-                $sMapFile = $sParentMapFile;
+        $isAdmin = isset($isAdmin) ? $isAdmin : $this->isAdmin();
+        $key = $languageIndex . ((int) $isAdmin);
+        if (!isset($this->_aLangMap[$key])) {
+            $this->_aLangMap[$key] = array();
+            $config = $this->getConfig();
+
+            $theme = $isAdmin ? 'admin' : $config->getConfigParam("sTheme");
+            $customTheme = $isAdmin ? 'admin' : $config->getConfigParam("sCustomTheme");
+            $languageAbbr = oxRegistry::getLang()->getLanguageAbbr($languageIndex);
+            $mapFiles = [
+                $config->getAppDir() . '/views/' . $theme . '/translations/map.php',
+                $config->getAppDir() . '/views/' . $theme . '/' . $languageAbbr . '/map.php',
+                $config->getAppDir() . '/views/' . $customTheme . '/translations/map.php',
+                $config->getAppDir() . '/views/' . $customTheme . '/' . $languageAbbr . '/map.php',
+            ];
+
+            foreach ($mapFiles as $tmpMapFile) {
+                if (file_exists($tmpMapFile) && is_readable($tmpMapFile)) {
+                    $mapFile = $tmpMapFile;
```

Comment:
Looks like we could reorder the $mapFiles and break the cycle on the first one we will find?

Reply:
yes good idea